### ISSUE TITLE
phpExtensions.phar: Not supported gzip compression and OpenSSL signatures

### ISSUE THREAD

**Issue by Izorkin** [NixOS/nixpkgs#89442]
Not supported gzip compression and OpenSSL signatures on phar package
```
phar version
PHP Version:               7.4.6
phar.phar version:         $Id: 341e5f3465f7f0cabb2fbde9d0551527ec7e8da3 $
Phar EXT version:          7.4.6
Phar API version:          1.1.1
Phar-based phar archives:  enabled
Tar-based phar archives:   enabled
ZIP-based phar archives:   enabled
gzip compression:          disabled
bzip2 compression:         disabled
supported signatures:      MD5, SHA-1, SHA-256, SHA-512
```
before update
```
PHP Version:               7.4.6
phar.phar version:         $Id: 341e5f3465f7f0cabb2fbde9d0551527ec7e8da3 $
Phar EXT version:          7.4.6
Phar API version:          1.1.1
Phar-based phar archives:  enabled
Tar-based phar archives:   enabled
ZIP-based phar archives:   enabled
gzip compression:          enabled
bzip2 compression:         disabled
supported signatures:      MD5, SHA-1, SHA-256, SHA-512, OpenSSL

```

cc @NixOS/php

**Comment by aanderse** [comment]
@Izorkin is this still an issue?

**Comment by Izorkin** [comment]
@aanderse yes, bzip2 compression disabled.
```
Phar: PHP Archive support  enabled
Phar API version           1.1.1
Phar-based phar archives   enabled
Tar-based phar archives    enabled
ZIP-based phar archives    enabled
gzip compression           enabled
bzip2 compression          disabled (install ext/bz2)
OpenSSL support            enabled
```

**Comment by Izorkin** [comment]
Working with this change:
```
diff --git a/pkgs/development/interpreters/php/7.4.nix b/pkgs/development/interpreters/php/7.4.nix
index b45da1fb86c..830f12deabf 100644
--- a/pkgs/development/interpreters/php/7.4.nix
+++ b/pkgs/development/interpreters/php/7.4.nix
@@ -9,7 +9,7 @@ let
   });

 in base.withExtensions ({ all, ... }: with all; ([
-  bcmath calendar curl ctype dom exif fileinfo filter ftp gd
+  bcmath bz2 calendar curl ctype dom exif fileinfo filter ftp gd
   gettext gmp iconv intl json ldap mbstring mysqli mysqlnd opcache
   openssl pcntl pdo pdo_mysql pdo_odbc pdo_pgsql pdo_sqlite pgsql
   posix readline session simplexml sockets soap sodium sqlite3

```

**Comment by Izorkin** [comment]
Found no ftps support:
```
FTP support    enabled
FTPS support   disabled
```

**Comment by Izorkin** [comment]
> Found no ftps support:
> 
> ```
> FTP support    enabled
> FTPS support   disabled
> ```

Fixed in #105808
